src/cpu/primitives.cc

```diff
@@ -804,6 +809,58 @@ namespace ctranslate2 {
     }
 #endif
 
+#ifdef CT2_WITH_RUY
+    case cpu::GemmBackend::RUY: {
+      static thread_local ruy::Context context;
+      context.set_max_num_threads(omp_get_max_threads());
```

Comment:
We should try setting the number of threads outside this function, if possible. I'm thinking:

* add function `get_ruy_context` in `src/cpu/backend.h` which returns the thread local Ruy context
* extend `set_num_threads` definition in `src/utils.cc` to also set the number of Ruy threads when compiling with Ruy

What do you think?

Reply:
Your plan sounds good! I was actually unsure about this design.